runtime/parser/lexer/lexer.go

```diff
@@ -49,6 +49,13 @@ type position struct {
 	column int
 }
 
+type LexerMode int
+
+const (
+	NORMAL = iota
+	STR_IDENTIFIER
```

Comment:
Suggestion: Maybe rename this to a more generic one like `stringInterpolation`, since more expressions will be supported in future.

btw, Go's constants also follow a Camel-case naming style (because starting with a upper-case letter means it becomes a global constant)